### ISSUE TITLE
attention trick implementation

### ISSUE THREAD

**Issue by kingfengji** [keras-team/keras#2525]
Hey guys! 
Inspired by the attention LSTM, 

Each time I have 3 input vectors, say x_1, x_2 and x_3. 

I wish to first make a linear combination layer_I = a_1_x1+a2_x2+a3*x3 

Then I merge this layer with some other sequencial layers.

I wish to learn the a_1 a_2 and a_3

How to do it in keras????

THANKS !!!!!!!!!!!!
with love

**Comment by braingineer** [comment]
Typically the attention is done with a single MLP that maps vectors to scores, then uses those scores in a softmax to get a probability distribution over the vectors.  Finally, you hadamard the probability distribution and initial vectors, sum over the sequence dimension (dimension 1). 

here's some code I wrote to do this.  
https://gist.github.com/braingineer/27c6f26755794f6544d83dec2dd27bbb

Though, you should definitely read up on attention further.  [Bahdanau et al](http://arxiv.org/abs/1409.0473) have a great paper on it.

**Comment by philipperemy** [comment]
Have a look at this:

https://github.com/philipperemy/keras-simple-attention-mechanism

It's a very simple Hello world attention mechanism but might address your needs!

**Comment by v1nc3nt27** [comment]
@braingineer Thanks for your code. Do you have a toy example on how to use it?

**Comment by dcabz** [comment]
@briangineer does your code include bahdanu's attention
